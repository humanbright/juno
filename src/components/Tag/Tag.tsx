import React, { forwardRef, HTMLAttributes, memo } from 'react';

import {
  RcBaseProps,
  RcPaletteProp,
  RcTheme,
  styled,
  withDeprecatedCheck,
  useThemeProps,
} from '../../foundation';
import { TagStyle } from './styles';
import { colorMap } from './utils';

type RcTagProps = {
  // TODO: remove colorMap
  /** tag color, default is 'highlight.b03' */
  color?: RcPaletteProp | keyof typeof colorMap;
  /** tag border color */
  borderColor?: RcPaletteProp;
  /** text color, default is 'neutral.f01' */
  textColor?: RcPaletteProp;
  /** The content of the tag. */
  children?: React.ReactNode;
  /** custom border radius for tag, default is `xl` */
  radius?: keyof RcTheme['radius'];
  /** @deprecated The content of the tag, please use children directly */
  content?: string;
} & RcBaseProps<HTMLAttributes<HTMLSpanElement>, 'color'>;

const _RcTag = memo(
  forwardRef<any, RcTagProps>((inProps, ref) => {
    const props = useThemeProps({ props: inProps, name: 'RcTag' });
    const {
      color,
      children,
      content,
      textColor,
      borderColor,
      radius,
      ...rest
    } = props;

    return (
      <span {...rest} ref={ref}>
        {content || children}
      </span>
    );
  }),
);

const RcTag = styled(
  withDeprecatedCheck(_RcTag, [
    {
      prop: 'content',
      comment: 'you should replace with children',
      time: '2021-1',
    },
  ]),
)`
  ${TagStyle}
`;

RcTag.defaultProps = {
  color: 'highlight.b03',
  textColor: 'neutral.f01',
  radius: 'xl',
};

RcTag.displayName = 'RcTag';

export { RcTag, RcTagProps };
