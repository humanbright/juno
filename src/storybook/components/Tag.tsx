import {
  getParsePaletteColor,
  palette2,
  radius,
  shadows,
  spacing,
  styled,
  typography,
} from '../../foundation';

export const Tag = styled.div`
  display: inline-flex;
  border-radius: ${radius('xxl')};
  overflow: hidden;
  box-shadow: ${shadows('4')};
  ${typography('caption1')};

  > * {
    margin: 0;
    padding: 0;
    padding: ${spacing(1, 2)};
  }

  > :first-child {
    background: ${({ color }) => getParsePaletteColor(color)};
  }

  > :last-child {
    background: ${palette2('neutral', 'b02')};
  }
`;

Tag.defaultProps = {
  color: 'success.b04',
};
