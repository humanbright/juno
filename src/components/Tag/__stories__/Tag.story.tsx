import { Meta, Story } from '@storybook/react';
import React, { ComponentProps } from 'react';

import {
  notShowInDocTable,
  paletteChoice,
  switchToControlKnobs,
} from '../../../storybook';
import { RcTag } from '../Tag';

export default {
  title: 'ðŸš€ Cleanup Components/Tag',
  component: RcTag,
  argTypes: {
    color: {
      control: {
        type: 'select',
        options: paletteChoice,
      },
    },
    textColor: {
      control: {
        type: 'select',
        options: paletteChoice,
      },
    },
    borderColor: {
      control: {
        type: 'select',
        options: paletteChoice,
      },
    },
  },
} as Meta;

type TagProps = ComponentProps<typeof RcTag>;

export const Tag: Story<TagProps> = ({ children, ...args }) => {
  switchToControlKnobs();
  return (
    <>
      <RcTag {...args}>{children}</RcTag>
    </>
  );
};

Tag.args = {
  children: 'tag test overflowsannsadsndjdnsaasddksjnadsndsasdasdadsadsdsa',
};

Tag.argTypes = {
  ...notShowInDocTable<keyof TagProps>(['content']),
};

Tag.parameters = {
  tags: [],
};

export const TagWithCustomBorder = () => {
  return (
    <RcTag
      color="neutral.b01"
      textColor="interactive.f01"
      borderColor="highlight.f02"
      onClick={() => {
        console.log('tag click');
      }}
    >
      tag test overflowsannsadsndjdnsaasddksjnadsndsasdasdadsadsdsa
    </RcTag>
  );
};
