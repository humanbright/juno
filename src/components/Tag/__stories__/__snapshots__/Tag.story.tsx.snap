// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots ðŸš€ Cleanup Components/Tag Tag 1`] = `
.c0 {
  display: inline-block;
  background-color: #FF8800;
  color: #FFFFFF;
  border-radius: 8px;
  text-align: center;
  padding: 0px 8px;
  font-size: 0.75rem;
  font-weight: 400;
  font-family: Lato,Helvetica,Arial,sans-serif;
  line-height: 16px;
  margin-left: 12px;
  min-width: 36px;
  max-width: 164px;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

<span
  className="c0"
>
  tag test overflowsannsadsndjdnsaasddksjnadsndsasdasdadsadsdsa
</span>
`;

exports[`Storyshots ðŸš€ Cleanup Components/Tag Tag With Custom Border 1`] = `
.c0 {
  display: inline-block;
  background-color: #FFFFFF;
  color: #066FAC;
  border: 1px solid #FF8800;
  border-radius: 8px;
  text-align: center;
  padding: 0px 8px;
  font-size: 0.75rem;
  font-weight: 400;
  font-family: Lato,Helvetica,Arial,sans-serif;
  line-height: 16px;
  margin-left: 12px;
  min-width: 36px;
  max-width: 164px;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

<span
  className="c0"
  onClick={[Function]}
>
  tag test overflowsannsadsndjdnsaasddksjnadsndsasdasdadsadsdsa
</span>
`;
